app/main.py

```diff
@@ -1,2 +1,114 @@
+from dataclasses import dataclass
+from typing import Any, Hashable, List, Optional, Iterator
+
+
+INITIAL_CAPACITY = 8
+LOAD_FACTOR = 0.75
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+    _hash: int
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table: List[Optional[Node]] = [None] * self.capacity
+
+    def calculate_index(self, key: Hashable, key_hash: int) -> int:
+        index = key_hash % self.capacity
+        while (
+                self.hash_table[index] is not None
+                and self.hash_table[index].key != key
+        ):
+            index = (index + 1) % self.capacity
+        return index
+
+    def resize(self) -> None:
+        old_hash_table = self.hash_table
+        self.capacity *= 2
+        self.hash_table = [None] * self.capacity
+        self.size = 0
```

Comment:
In the `resize` method, resetting `self.size` to zero is incorrect. The size should reflect the number of elements in the hash table after rehashing, so it should not be reset here.